app/controllers/api/v1/articles_controller.rb

```diff
@@ -1,8 +1,15 @@
+# frozen_string_literal: true
+
 class Api::V1::ArticlesController < ApplicationController
   rescue_from ActiveRecord::RecordNotFound, with: :render_active_record_error
+  before_action :check_valid_category, only: :index, if: :searching_for_categoriezed_content
 
   def index
-    articles = Article.all
+    articles = if searching_for_categoriezed_content
+                 Article.where(category: params['category'])
+               else
+                 Article.all
+               end
```

Comment:
I would change it to 
```
if selecting_category
  articles = Article.where(category: params['category'])
else 
articles = Article.all
end
```

its a bit more readable in my opinion. And the naming of the method is a bit out of place because you are not make a search, you are choosing or selecting a category

Reply:
Yes! That is a very good opinion, we totally agree and have changed it now!